### DISCUSSION TITLE
MP-QUIC是否支持1:1的双路径冗余发送？应用可以客制化不同路径上的发送策略么？草案中提到的QoE_CONTROL_SIGNAL有应用示例么？

### DISCUSSION THREAD

**Discussion by AskaMgl** [alibaba/xquic#282] [Q&A]
如题

**Comment by ZephyrPhoenix** [comment]
你好！所以xquic实现了冗余发送吗？

**Comment by Ya-Pasha-364shy** [comment]
Hi everyone! I spent some time looking into how sending data by paths works in XQUIC and I can tell you that **no**, sending backup data by paths is not supported here. Instead, only lost packets are retransmitted.

Look at the `xqc_conn_retransmit_lost_packets_batch` or `xqc_conn_retransmit_lost_packets` methods in `xqc_conn.c`, which send packets that are in the doubly linked list of the appropriate type by all paths.

in fact, packets are marked as lost using `xqc_send_ctl.c` (method` xqc_send_ctl_detect_lost`, triggered by a timer), which checks whether there are lost packets in the queue of unanswered packets and, if there are, then a move is made to the appropriate queue of lost packets.

I hope, that i helped u in your question
